src/pages/journal/components/vehicle-details/__tests__/vehicle-details.spec.ts

```diff
@@ -0,0 +1,62 @@
+import { TestBed, async, ComponentFixture } from '@angular/core/testing';
+import { IonicModule } from 'ionic-angular';
+import { VehicleDetailsComponent } from '../vehicle-details';
+import { By } from '@angular/platform-browser';
+
+
+describe('VehicleDetailsComponent', () => {
+  let fixture: ComponentFixture<VehicleDetailsComponent>;
+  let component: VehicleDetailsComponent;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        VehicleDetailsComponent
+      ],
+      imports: [IonicModule],
+
+    }).compileComponents().then(() => {
+      fixture = TestBed.createComponent(VehicleDetailsComponent);
+      component = fixture.componentInstance;
+    })
+  }));
+
+  describe('Class', () => {
+    it('should create', () => {
+      expect(component).toBeDefined();
+    });
+  });
+
+  describe('DOM', () => {
+    it('should show vehicle details when showVehicleDetails is true', () => {
+      component.showDimensions = true;
+      fixture.detectChanges();
+      const ionRows = fixture.debugElement.queryAll(By.css('ion-row'));
+      expect(ionRows).toHaveLength(2);
+    });
+
+    it('should not vehicle details when showVehicleDetails is false', () => {
+      component.showDimensions = false;
+      fixture.detectChanges();
+      const ionRows = fixture.debugElement.queryAll(By.css('ion-row'));
+      expect(ionRows).toHaveLength(1);
+    });
+
+    it('should show number of seats when showNumberOfSeats is true', () => {
+      component.showDimensions = false;
+      component.showNumberOfSeats = true;
+      fixture.detectChanges();
+      const ionCols = fixture.debugElement.queryAll(By.css('ion-col'));
+      expect(ionCols).toHaveLength(2)
+    });
+
+    it('should not show number of seats when showNumberOfSeats is false', () => {
+      component.showDimensions = false;
+      component.showNumberOfSeats = true;
```

Comment:
Not doing what the test name says here - seems identical to the test above.

Reply:
fixed - that's poor copy and paste work